consul crash with ports config parameters

I found when I used below configs (Consul 1.0.0/1.0.1/1.0.2), consul will crash
......
   "ports": {
        "serf_wan": -1
    },
  ......

but with 0.9.3 it is ok.

```
consul agent -dev -config-file=crash.json
==> Starting Consul agent...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x4b2615e]

goroutine 1 [running]:
github.com/hashicorp/consul/agent.(*Agent).consulConfig(0xc4200781a0, 0xc42024d450, 0xdf8475800, 0xc42018e360)
        /Users/james/projects/_go/src/github.com/hashicorp/consul/agent/agent.go:701 +0x32e
github.com/hashicorp/consul/agent.(*Agent).Start(0xc4200781a0, 0xc4200781a0, 0x0)
        /Users/james/projects/_go/src/github.com/hashicorp/consul/agent/agent.go:276 +0x228
github.com/hashicorp/consul/command/agent.(*cmd).run(0xc4202b9800, 0xc420010160, 0x2, 0x2, 0x0)
        /Users/james/projects/_go/src/github.com/hashicorp/consul/command/agent/agent.go:337 +0x414
github.com/hashicorp/consul/command/agent.(*cmd).Run(0xc4202b9800, 0xc420010160, 0x2, 0x2, 0xc4202c16e0)
        /Users/james/projects/_go/src/github.com/hashicorp/consul/command/agent/agent.go:77 +0x50
github.com/hashicorp/consul/vendor/github.com/mitchellh/cli.(*CLI).Run(0xc42018d320, 0xc42018d320, 0x40, 0xc4202c1900)
        /Users/james/projects/_go/src/github.com/hashicorp/consul/vendor/github.com/mitchellh/cli/cli.go:242 +0x1eb
main.realMain(0x4bbb147)
        /Users/james/projects/_go/src/github.com/hashicorp/consul/main.go:52 +0x416
main.main()
        /Users/james/projects/_go/src/github.com/hashicorp/consul/main.go:19 +0x22
```

--- a/agent/config/builder.go
+++ b/agent/config/builder.go
@@ -369,6 +369,9 @@ func (b *Builder) Build() (rt RuntimeConfig, err error) {
 	if ipaddr.IsAny(b.stringVal(c.AdvertiseAddrWAN)) {
 		return RuntimeConfig{}, fmt.Errorf("Advertise WAN address cannot be 0.0.0.0, :: or [::]")
 	}
+	if serfPortWAN < 0 {
+		return RuntimeConfig{}, fmt.Errorf("ports.serf_wan must be a valid port from 1 to 65535")
+	}
 
 	bindAddr := bindAddrs[0].(*net.IPAddr)
 	advertiseAddr := b.makeIPAddr(b.expandFirstIP("advertise_addr", c.AdvertiseAddrLAN), bindAddr)

--- a/agent/config/runtime_test.go
+++ b/agent/config/runtime_test.go
@@ -1065,6 +1065,23 @@ func TestConfigFlagsAndEdgecases(t *testing.T) {
 			},
 		},
 		{
+			desc: "serf wan port > 0",
+			args: []string{`-data-dir=` + dataDir},
+			json: []string{`{
+				"ports": {
+					"serf_wan": -1
+				},
+				"advertise_addr_wan": "1.2.3.4"
+			}`},
+			hcl: []string{`
+				ports {
+					serf_wan = -1
+				}
+				advertise_addr_wan = "1.2.3.4"
+			`},
+			err: "ports.serf_wan must be a valid port from 1 to 65535",
+		},
+		{
 			desc: "serf bind address lan template",
 			args: []string{`-data-dir=` + dataDir},
 			json: []string{`{ "serf_lan": "{{ printf \"1.2.3.4\" }}" }`},

